### PR TITLE
Fixed critical error on placeOrder function + added websockets

### PR DESCRIPTION
**Pull request by saniales**

### CODE REVIEW

File: hitbtc.go [hunk 1]
```diff
@@ -286,18 +290,28 @@ func (b *HitBtc) PlaceOrder(requestOrder Order) (responseOrder Order, err error)
...
 	if err != nil {
 		return
 	}
+
 	var response interface{}
 	if err = json.Unmarshal(r, &response); err != nil {
 		return
-	}
-	if err = handleErr(response); err != nil {
+	} else if err = handleErr(response); err != nil {
```
Comment (bitbandi) [inline hitbtc.go]: if the above branch always returns, no need the "else", just "if"

Comment (saniales) [reply hitbtc.go]: Thx will correct

**Comment by saniales** [comment]
corrected

**Comment by saniales** [comment]
Are you going to merge this?

File: ticker.go [hunk 2]
```diff
@@ -18,15 +19,26 @@ type Ticker struct {
...
 
+// Tickers rapresents a set of a valid Tickers struct
+type Tickers []Ticker
+
 func (t *Ticker) UnmarshalJSON(data []byte) error {
 	var err error
 	type Alias Ticker
 	aux := &struct {
 		Timestamp string `json:"timestamp"`
+		High      string `json:"high"`
```
Comment (bitbandi) [inline ticker.go:30]: This is for when high/low is missing?

File: ticker.go [hunk 3]
```diff
@@ -36,3 +48,26 @@ func (t *Ticker) UnmarshalJSON(data []byte) error {
...
+
+	if err := json.Unmarshal(data, &arr); err != nil {
+		return err
+	}
+
+	for _, v := range arr {
+		var ticker Ticker
+
+		err := json.Unmarshal(v, &ticker)
+		if ticker.Volume == 0 {
```
Comment (bitbandi) [inline ticker.go:63]: sould not filter out if volume is zero

Comment (saniales) [reply ticker.go:63]: If volume is 0 the market is inactive
This is why it is filtered out

File: ticker.go (hunk 2 above)
Comment (saniales) [reply ticker.go:30]: Yes

File: ticker.go (hunk 3 above)
Comment (bitbandi) [reply ticker.go:63]: inactive means, "disabled" by hitbtc, or no trades on that day?

Comment (saniales) [reply ticker.go:63]: Inactive, I checked it by running tests

Comment (saniales) [reply ticker.go:63]: Sorry disabled

File: ticker.go (hunk 2 above)
Comment (bitbandi) [reply ticker.go:30]: This can be handled with "omitempty" in ticker struct

File: ticker.go (hunk 3 above)
Comment (bitbandi) [reply ticker.go:63]: hmm, than, this may hide the "enabled but no trades today" market from list :/

**Comment by bitbandi** [comment]
I imported some codes from this PR, but leave leave the ticker filterout yet (need recheck that, i dont want to filter out the "enabled but no trades today" markets.

**Comment by saniales** [comment]
Ok

**Comment by saniales** [comment]
I see you imported my code, but I am not set as author of the commits, please allow me to be a contributor for next commits, thanks :)

**Comment by saniales** [comment]
I will now close this, thank you

**Pull request closed without merging**